src/noob/yaml.py

```diff
@@ -257,10 +275,26 @@ def _complete_header(cls: type[Self], data: dict, file_path: str | Path) -> dict
             logger.debug(data)
 
             header = cls._yaml_header(data)
-            data = {**header, **data}
+            comment: None | list[CommentToken] = None
+            for i, (key, value) in enumerate(header.items()):
+                if key in data:
+                    # pop it, preserving comments that start on following lines
+                    # to re-inject after the header block, if present
+                    if key in data.ca.items and data.ca.items[key][2].value.startswith("\n"):
+                        if comment is None:
+                            comment = data.ca.items.pop(key)
+                        else:
+                            comment[2].value += data.ca.items.pop(key)[2].value
+                    del data[key]
+                data.insert(i, key, value)
+            if comment:
+                # insert newline comments after noob_version,
+                # which is the last key in the header block
+                data.ca.items["noob_version"] = comment
```

Comment:
comments are attached to the most recent data line, so e.g. see below in the tests with "some comment that should be preserved" - that gets attached to `noob_version`. 

however we *do* want to keep comments that are just inline comments for that specific line. 

so we remove any comments that start with newlines and append them after `noob_version`, but keep the rest.